srml/support/src/dispatch.rs

```diff
@@ -67,38 +67,114 @@ pub trait Parameter: Codec + Clone + Eq {}
+/// 
 /// decl_module! {
-///		pub struct Module<T: Trait> for enum Call where origin: T::Origin
-///{}
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+/// 		fn my_function(origin, var: u64) -> Result {
+///				// Your implementation
+///				Ok(())
+/// 		}
+///
+///			// Public functions are easily accessible to other modules
+///			pub fn my_public_function(origin) -> Result {
+///				Ok(())
+/// 		}
+///		}
 ///	}
+/// # fn main() {}
+/// ```
+///
+/// The declaration is set with the header where:
+///
+/// * `Module`: The struct generated by the macro, with type `Trait`.
+/// * `Call`: The enum generated for every module, which implements [`Callable`](./dispatch/trait.Callable.html).
+/// * `origin`: Alias of `T::Origin`, declared by the [`impl_outer_origin!`](./macro.impl_outer_origin.html) macro.
+/// * `Result`: The expected return type from module functions.
+///
+/// ## Shorthand Example
+///
+///	```rust
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
+/// 
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+/// 		fn my_function(origin) {
+///				// Your implementation
+/// 		}
+///		}
+/// }
+/// # fn main() {}
+/// ```
+///
+/// The macro automatically expands a shorthand function declaration to return the `Result` type.
 ///
-/// where "Trait" is a trait describing this module's requirements for the Runtime type.
-/// T::Origin is declared using a impl_outer_origin! per-module macro (which is generated by the
-/// construct_runtime! macro) and automatically includes all the modules that are used by the
-/// runtime (alongside with a variant called "system").
+/// ## Priviledged Function Example
 ///
-/// A runtime module is a collection of functions unified by a common problem domain and certain
-/// shared types. The "functions" do not actually return values (see Dispatchable) and are only
-/// used for side effects.
+/// ```rust
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
 ///
-/// For every module an associated enum (usually "Call") is generated with every variant
-/// corresponding to a function of the module. This enum implements Callable and thus its values
-/// can be used as an extrinsic's payload.
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+///			fn my_function() -> Result {
+///				// Your implementation
+/// 			Ok(())
+/// 		}
+///		}
+/// }
+/// # fn main() {}
+/// ```
 ///
-/// The `on_initialize` and `on_finalize` functions are special, since it can either take no
-/// parameters, or one parameter, which has the runtime's block number type.
+/// If the `origin` param is omitted, the macro assumes the `origin` to be `Root`.
```

Comment:
We can be a little more explicit here.

`origin` is not assumed to be `Root`, we add:

```
ensure_root(origin)?;
```

as the first line of the function, which means if the origin is not root, the function will immediately return an error and none of the function will run.

```suggestion
/// If the `origin` param is omitted, the macro adds it as the first parameter and adds `ensure_root(origin)` as the first line of the function.
```